api/mapping/services_rules.py

```diff
@@ -841,3 +842,141 @@ def remove_mapping_rules(request, scan_report_id):
+
+            # For each descendant found
+            for descendant in descendants:
+                # get the concept id
+                desc = descendant.descendant_concept_id
+                # Check if the descendant is a mapped concept in any other scan report
+                # and that it's not the same as the original concept
+                if (desc in all_mapping_rules) and (desc != rule):
+
+                    # Get all the details for that descendant:
+                    # descendant name, current mapping rule name, where that descendant is mapped to
+                    desc_name, rule_name, source_ids = get_concept_details(rule, desc)
+                    # Append all descendant details to a list
+                    descendant_list.append(
+                        {
+                            "d_id": desc,
+                            "d_name": desc_name,
+                            "source": source_ids,
+                        }
+                    )
+            # For each ancestor found
+            for ancestor in ancestors:
+                # get the concept_id
+                anc = ancestor.ancestor_concept_id
+                # If ancestor is a mapping rule in any other scan report,
+                # and is not the same rule as the current one
+                if (anc in all_mapping_rules) and (anc != rule):
+
+                    # Get all the details for that ancestor:
+                    # ancestor name, current mapping rule name, where that ancestor is mapped to
+                    concept, rule_name, source_ids = get_concept_details(rule, anc)
+                    # Append all the ancestor details to a list
+                    ancestors_list.append(
+                        {
+                            "a_id": anc,
+                            "a_name": concept,
+                            "source": source_ids,
+                        }
+                    )
+
+            # Append all the descendants/ancestors of the current mapping rule in a dict
+            # Do not append if both lists are empty
+            if ancestors_list or descendant_list:
+                data.append(
+                    {
+                        "rule_id": rule,
+                        "rule_name": rule_name,
+                        "anc_desc": [
+                            {
+                                "descendants": descendant_list,
+                                "ancestors": ancestors_list,
+                            }
+                        ],
+                    }
+                )
+            else:
+                raise Exception(
+                    "No ancestors or descendants of {} appear in any other Scan Reports".format(
+                        rule_name
+                    )
+                )
+            # Reset lists before moving on to the next mapping rule
+            descendant_list = []
+            ancestors_list = []
+        except Exception as e:
```

Comment:
I think this will need changing, because `except Exception` will be triggered by other exceptions that inherit from `Exception`, like `KeyError` or `ArithmeticError`, so this is no better than a bare `except`. 

I think better would be to just add a `print` statement in the else clause, and remove the try-except. As written now, _any_ error would trigger the error message, which would be possibly misleading if it's really a `KeyError`or something else triggering it.